builtInFunctions/esdtMetaDataRecreate.go

```diff
@@ -141,7 +141,7 @@ func getEsdtInfo(
 	if err != nil {
 		return nil, err
 	}
-	if core.IsDynamicESDT(tokenType) {
+	if tokenType == uint32(core.DynamicSFT) || tokenType == uint32(core.DynamicMeta) {
 		esdtData, err := storageHandler.GetMetaDataFromSystemAccount(esdtTokenKey, nonce)
```

Comment:
there is a possibility to have this nil. 

Contract creates DynamicSFT and sends cross shard all the liquidity. 

After it wants to recreate the data - it should work.

Reply:
This should not happen. On transfer we have this check 
`keepMetadataOnZeroLiquidity, err := hasDynamicRole(acntSnd, tickerID, e.marshaller)` which will prevent this case

Reply:
Will add a test to be sure

Reply:
There might be another case here. The update role is given to another shard before any DynSFT is transfered to that shard. Recreate will not work because no data is saved on that system account. 
Will add test and fix :hammer_and_wrench: